operators/video_streaming/UNIT_TESTS_SUMMARY.md

```diff
@@ -0,0 +1,207 @@
+# C++ Unit Tests for Video Streaming Operators - Summary
+
+## Overview
+
+This document summarizes the C++ unit tests for the `video_streaming` operators, providing comprehensive coverage of all client and server components.
+
+## Test Suites
+
+This package includes **31 unit tests** across all video streaming operators:
+
+### 1. **StreamingClientOp Tests** (13 tests)
+   - Basic initialization
+   - Parameter validation (video, network, streaming modes)
+   - Frame validation parameters
+   - Edge cases and boundaries
+   - Resource management
+
+### 2. **StreamingServerResource Tests** (8 tests)
+   - Resource creation and configuration
+   - Various resolutions and frame rates
+   - Port configurations
+   - Streaming direction settings
+   - Multi-instance mode
+
+### 3. **StreamingServerUpstreamOp Tests** (6 tests)
+   - Operator initialization
+   - Custom video parameters
+   - Setup and cleanup
+   - Shared resource configuration
+   - Multiple operators with shared resource
+
+### 4. **StreamingServerDownstreamOp Tests** (4 tests)
+   - Operator initialization
+   - Custom video parameters
+   - Setup and cleanup
+   - Resource cleanup
+
+## File Structure
+
+```
+operators/video_streaming/
+├── video_streaming_client/
+│   ├── tests/
+│   │   ├── test_video_streaming_client_op.cpp # Client unit tests (13 tests)
+│   │   ├── CMakeLists.txt                     # Build config
+│   │   └── README.md                          # Documentation
+│   └── CMakeLists.txt                         # Updated to include tests
+│
+└── video_streaming_server/
+    ├── tests/
+    │   ├── test_video_streaming_server_ops.cpp # Server unit tests (18 tests)
+    │   ├── CMakeLists.txt                     # Build config
+    │   └── README.md                          # Documentation
+    └── CMakeLists.txt                         # Updated to include tests
+```
+
+## Test Statistics
+
+| Component | Tests | Lines of Code | Coverage | Execution Time |
+|-----------|-------|---------------|----------|----------------|
+| StreamingClientOp | 13 | ~700 | Initialization, parameters, setup, edge cases | ~0.06 sec |
+| StreamingServerResource | 8 | ~300 | Resource creation, configuration, various settings | (included below) |
+| StreamingServerUpstreamOp | 6 | ~250 | Initialization, setup, cleanup, shared resources | (included below) |
+| StreamingServerDownstreamOp | 4 | ~150 | Initialization, setup, cleanup | (included below) |
+| **Total** | **31** | **~1,400** | **Comprehensive** | **~0.12 sec** |
```

Comment:
These statistics can easily get out of sync when adding more tests. Could you please remove them?